collect_app/src/main/java/org/odk/collect/android/widgets/AudioWidget.java

```diff
@@ -101,24 +98,33 @@ public AudioWidget(Context context, QuestionDetails prompt, WaitingForDataRegist
         this.waitingForDataRegistry = waitingForDataRegistry;
         this.mediaManagerListener = mediaManagerListener;
 
-        captureButton = createSimpleButton(getContext(), R.id.capture_audio, getFormEntryPrompt().isReadOnly(), getContext().getString(R.string.capture_audio), getAnswerFontSize(), this);
-
-        chooseButton = createSimpleButton(getContext(), R.id.choose_sound, getFormEntryPrompt().isReadOnly(), getContext().getString(R.string.choose_sound), getAnswerFontSize(), this);
-
-        // finish complex layout
-        LinearLayout answerLayout = new LinearLayout(getContext());
-        answerLayout.setOrientation(LinearLayout.VERTICAL);
-        answerLayout.addView(captureButton);
-        answerLayout.addView(chooseButton);
-        answerLayout.addView(audioController);
-        addAnswerView(answerLayout, WidgetViewUtils.getStandardMargin(context));
-
         hideButtonsIfNeeded();
 
         binaryName = questionDetails.getPrompt().getAnswerText();
         updatePlayerMedia();
     }
 
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        binding = AudioWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
+
+        if (prompt.isReadOnly()) {
+            binding.captureButton.widgetButton.setVisibility(View.GONE);
+            binding.chooseButton.widgetButton.setVisibility(View.GONE);
+        } else {
+            binding.captureButton.widgetButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+            binding.chooseButton.widgetButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+
+            binding.captureButton.widgetButton.setText(getContext().getString(R.string.capture_audio));
```

Comment:
Feels like the button text should just live in a layout?

Reply:
I use 
`
    <include
        android:id="@+id/capture_button"
        layout="@layout/widget_button"/>

    <include
        android:id="@+id/choose_button"
        layout="@layout/widget_button"/>`

in audio_widget.xml file. So I have added the button text here.

Reply:
I think I'd be happier duplicating the layout here than having hardcoded text set programmatically. My thinking is that it's good to be able to see the layout as it will appear (with the text) in the design preview for it.

Reply:
For the other file widgets, I was using the same `widget_button.xml` layout. So, I should have a separate layout for all of them too?

Reply:
I think instead of sharing an `include` maybe we could create a custom button class with the shared values. Or, create a style that can be shared. These would let us avoid duplication but avoid having to set the text programmatically (which I'd like to avoid as much as possible).